### PR TITLE
Deep Learning from Scikit-Learn

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

New feature, #434 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
Validation results:
```
-----------------------------------------------------------------------
                           Failure Summary
-----------------------------------------------------------------------

Validated: 486 Failed: 0 Run: 486
```

**Comment by wangcj05** [comment]
Two new tests are added, and the test results:
```
(raven_libraries) [wangc][~/projects/wangc/raven]> ./run_framework_tests -j6 --re=framework.ROM.SKLearn.MLP
/Users/wangc/miniconda2/envs/raven_libraries/bin/conda
framework/ROM/SKLearn.MLPR................................................................................ OK
framework/ROM/SKLearn.MLPC................................................................................ OK
-------------------------------------------------------------------------------------------------------------
Ran 2 tests in 7.1 seconds
2 passed, 0 skipped, 0 pending, 0 failed
```

**Comment by wangcj05** [comment]
The code changes should not affect the current branch and tests.

**Comment by wangcj05** [comment]
@PaulTalbot-INL @alfoa This PR is ready for you to review.

**Comment by wangcj05** [comment]
In addition, this PR requires the version of Scikit-Learn >= 18.0.

File: doc/user_manual/model.tex [hunk 1]
```diff
@@ -655,6 +655,204 @@ \section{Models}
 }
 %%%%%%%%%%%%%%%%%%%% End Common Gaussian Process Parameters %%%%%%%%%%%%%%%%%%%%
 
+%%%%%%%%%%%%%%%%%%%%%% Common Multi-layer Perceptron Parameters %%%%%%%%%%%%%%%%%%%%%%
+\newcommand{\hiddenLayerSizesMLPDescription}[1]
+{
+  \xmlNode{hidden\_layer\_sizes}, \xmlDesc{tuple, length = n\_layers - 2, optional field}, the ith
+  element represents the number of neutrons in the ith hidden layer.
```
Comment (PaulTalbot-INL) [inline doc/user_manual/model.tex]: are `neutrons` the same as `neurons`?

File: doc/user_manual/rom.tex [hunk 2]
```diff
@@ -2804,6 +2804,105 @@ \subsubsection{SciKitLearn}
 </Simulation>
 \end{lstlisting}
 
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ %%%%% ROM Model - SciKitLearn: Neural Network Models %%%%%%%
+\paragraph{Neural Network Models}
+\label{DNN}
+It has been more than 70 years since Warren McCulloch and Water Pitts modeled the first
+artificial neural network (ANN) that mimicked the way brains work. These day, deep learning
```
Comment (PaulTalbot-INL) [inline doc/user_manual/rom.tex]: "These days"

File: doc/user_manual/rom.tex [hunk 3]
```diff
@@ -2804,6 +2804,105 @@ \subsubsection{SciKitLearn}
...
+%
+\textbf{Multi-layer Perceptron (MLP)} is a supervised learning algorithm that can learn
+a non-linear function approximator for either classifcation or regression. It is different
+from logistic regression, in that between the input and output layer, there can be one
+or more non-linear layers, called hidden layers.
+%
+The advantages of Multi-layer Perceptron are:
+\begin{itemize}
+  \item Capability to learn non-linear models
+  \item Capability to learn models in real-time (on-line learning)
```
Comment (PaulTalbot-INL) [inline doc/user_manual/rom.tex]: "online"

File: doc/user_manual/rom.tex [hunk 4]
```diff
@@ -2804,6 +2804,105 @@ \subsubsection{SciKitLearn}
...
+  \item \validationFractionMLPDescription{0.1}
+  \item \betaAMLPDescription{0.9}
+  \item \betaBMLPDescription{0.999}
+  \item \epsilonMLPDescription{1e-8}
+\end{itemize}
+
+%%%%%%%%%%%%%%%
+\subparagraph{MLPRegressor}
+\mbox{}
+\\The \textit{MLPRegressor} implements a multi-layer perceptron algorithm that trains using \textbf{Backpropagation} with
```
Comment (PaulTalbot-INL) [inline doc/user_manual/rom.tex]: Is "backpropagation" a key word, a node name, a node string, or a name?  It's not clear to me.  If none of the above, should it be "back propagation"?

File: doc/user_manual/model.tex (hunk 1 above)
Comment (wangcj05) [reply doc/user_manual/model.tex]: fixed. Should be ```neurons```

File: doc/user_manual/rom.tex (hunk 2 above)
Comment (wangcj05) [reply doc/user_manual/rom.tex]: fixed.

File: doc/user_manual/rom.tex (hunk 3 above)
Comment (wangcj05) [reply doc/user_manual/rom.tex]: fixed.

File: doc/user_manual/rom.tex (hunk 4 above)
Comment (wangcj05) [reply doc/user_manual/rom.tex]: It is a word generally used in the neuron network.

Comment (PaulTalbot-INL) [reply doc/user_manual/rom.tex]: As a single word?  Interesting.

**Pull request merged**